### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by RomaSerenkov**

### CODE REVIEW

File: migrations/Version20210118135848.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20210118135848 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return '';
```
Comment (artsems) [inline migrations/Version20210118135848.php]: Why there is a method which returns an empty string?

File: migrations/Version20210118135848.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+declare(strict_types=1);
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
```
Comment (artsems) [inline migrations/Version20210118135848.php]: Remove or rewrite comment.

File: src/Controller/SecurityController.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+class SecurityController extends AbstractController
+{
+    /**
+     * @Route("/login", name="app_login")
+     */
+    public function login(AuthenticationUtils $authenticationUtils): Response
+    {
+        // if ($this->getUser()) {
+        //     return $this->redirectToRoute('target_path');
```
Comment (artsems) [inline src/Controller/SecurityController.php]: Remove commented code.

File: src/Controller/SecurityController.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Authentication\AuthenticationUtils;
+
+class SecurityController extends AbstractController
+{
+    /**
+     * @Route("/login", name="app_login")
```
Comment (artsems) [inline src/Controller/SecurityController.php]: Rename route as `security_login`.

File: src/Controller/TestController.php [hunk 5]
```diff
@@ -10,10 +10,10 @@
 class TestController extends AbstractController
 {
     /**
-     * @Route("/hello")
+     * @Route("/hello", name="hello")
```
Comment (artsems) [inline src/Controller/TestController.php]: Rename to `test_hello`.

File: src/Controller/SecurityController.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        // get the login error if there is one
+        $error = $authenticationUtils->getLastAuthenticationError();
+        // last username entered by the user
+        $lastUsername = $authenticationUtils->getLastUsername();
+
+        return $this->render('security/login.html.twig', ['last_username' => $lastUsername, 'error' => $error]);
+    }
+
+    /**
+     * @Route("/logout", name="app_logout")
```
Comment (artsems) [inline src/Controller/SecurityController.php]: Rename route as `security_logout`.

File: src/Entity/User.php [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    public function setEmail(string $email): self
+    {
+        $this->email = $email;
+
+        return $this;
+    }
+
+    /**
```
Comment (artsems) [inline src/Entity/User.php]: Remove all getters\setters annotations here and below in this file.

File: src/Repository/UserRepository.php [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        if (!$user instanceof User) {
+            throw new UnsupportedUserException(sprintf('Instances of "%s" are not supported.', \get_class($user)));
+        }
+
+        $user->setPassword($newEncodedPassword);
+        $this->_em->persist($user);
+        $this->_em->flush();
+    }
+
+    // /**
```
Comment (artsems) [inline src/Repository/UserRepository.php]: Remove commented code.

File: src/Repository/UserRepository.php [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    /**
+     * Used to upgrade (rehash) the user's password automatically over time.
+     */
+    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    {
+        if (!$user instanceof User) {
+            throw new UnsupportedUserException(sprintf('Instances of "%s" are not supported.', \get_class($user)));
+        }
+
+        $user->setPassword($newEncodedPassword);
```
Comment (artsems) [inline src/Repository/UserRepository.php:34]: Add a newline after `setPassword` (just makes it more readable.

Comment (artsems) [inline src/Repository/UserRepository.php:34]: Add a newline after `setPassword` (just makes code more readable).

Comment (artsems) [inline src/Repository/UserRepository.php:34]: Add a newline after `setPassword` (just makes code more readable).

File: src/Security/LoginFormAuthenticator.php [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Component\Security\Guard\Authenticator\AbstractFormLoginAuthenticator;
+use Symfony\Component\Security\Guard\PasswordAuthenticatedInterface;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class LoginFormAuthenticator extends AbstractFormLoginAuthenticator implements PasswordAuthenticatedInterface
+{
+    use TargetPathTrait;
+
+    public const LOGIN_ROUTE = 'app_login';
```
Comment (artsems) [inline src/Security/LoginFormAuthenticator.php]: Remember about renamed route.

Comment (artsems) [inline src/Security/LoginFormAuthenticator.php]: Remember about renamed route.

**Review by artsems — CHANGES_REQUESTED** [review]

File: migrations/Version20210118135848.php (hunk 1 above)
Comment (RomaSerenkov) [reply migrations/Version20210118135848.php]: Resolve

File: migrations/Version20210118135848.php (hunk 2 above)
Comment (RomaSerenkov) [reply migrations/Version20210118135848.php]: Resolve

File: src/Controller/SecurityController.php (hunk 3 above)
Comment (RomaSerenkov) [reply src/Controller/SecurityController.php]: Resolve

File: src/Controller/SecurityController.php (hunk 4 above)
Comment (RomaSerenkov) [reply src/Controller/SecurityController.php]: Resolve

File: src/Controller/TestController.php (hunk 5 above)
Comment (RomaSerenkov) [reply src/Controller/TestController.php]: Resolve

File: src/Controller/SecurityController.php (hunk 6 above)
Comment (RomaSerenkov) [reply src/Controller/SecurityController.php]: Resolve

File: src/Entity/User.php (hunk 7 above)
Comment (RomaSerenkov) [reply src/Entity/User.php]: Resolve

File: src/Repository/UserRepository.php (hunk 8 above)
Comment (RomaSerenkov) [reply src/Repository/UserRepository.php]: Resolve

File: src/Repository/UserRepository.php (hunk 9 above)
Comment (RomaSerenkov) [reply src/Repository/UserRepository.php:34]: Resolve

Comment (RomaSerenkov) [reply src/Repository/UserRepository.php:34]: Resolve

Comment (RomaSerenkov) [reply src/Repository/UserRepository.php:34]: Resolve

File: src/Security/LoginFormAuthenticator.php (hunk 10 above)
Comment (RomaSerenkov) [reply src/Security/LoginFormAuthenticator.php]: Resolve

Comment (RomaSerenkov) [reply src/Security/LoginFormAuthenticator.php]: Resolve

**Review by artsems — APPROVED** [review]

**Pull request merged**